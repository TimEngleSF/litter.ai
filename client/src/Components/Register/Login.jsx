import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../../utils/UserContext';
import { fetchLogin, fetchLogOut } from '../../utils/fetchUserData';

export default function Login({ setUser, user }) {
  const userContext = useContext(UserContext);
  console.log('userContext: ', userContext);
  const [loginData, setLoginData] = useState({
    email: '',
    password: '',
  });

  const handleLogin = async () => {
    try {
      await userContext.login(loginData);
    } catch (err) {
      console.error(err);
    }
    // try {
    //   const userData = await fetchLogin(loginData);
    //   await setUser(userData);
    // } catch (err) {
    //   console.error(err);
    // }
  };

  const handleChange = (e) => {
    e.preventDefault();
    setLoginData({ ...loginData, [e.target.id]: e.target.value });
  };

  const handleCheck = () => {
    console.log('user: ', user);
  };

  return (
    <div>
      <form>
        <label htmlFor="email">
          email:
          <input
            type="text"
            id="email"
            value={loginData.email}
            onChange={(e) => handleChange(e)}
          />
        </label>
        <label htmlFor="password">
          password:
          <input
            type="text"
            id="password"
            value={loginData.password}
            onChange={(e) => handleChange(e)}
          />
        </label>
        <button type="button" onClick={handleLogin}>Login</button>
        <button type="button" onClick={handleCheck}>Test Login Data</button>
        <button type="button" onClick={() => fetchLogOut()}>Log out</button>
        <Link to="/leaderboard"><button type="button">Leaderboard</button></Link>
      </form>
    </div>
  );
}
