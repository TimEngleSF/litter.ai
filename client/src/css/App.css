:root {
  --main-green: #076B49;
  --main-green-half: #076b49c9;
  --sec-green: #74CC67;
  --black: #000;
  --white: #FFF;
  --gray: #ededed;
  --dark-gray: #7c7c7c;
  --lb-gray: #E0E9E6;
  --shadow-color: hsl(0deg 0% 0% / 26%);
}

@media only screen and (max-width: 767px) {
  .App {
    grid-template-columns: 1fr;
  }
}

@media only screen and (min-width: 768px) {
  .App {
    grid-template-columns: 2fr 3fr 2fr;
  }
}

.App {
  width: 100dvw;
  max-width: 100%;
  height: 100dvh;
  display: grid;
  grid-template-rows: repeat(10, 1fr);
  justify-items: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100dvh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* Buttons */

.lower-buttons {
  max-width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 1rem;
}

.lower-buttons button {
  width: 85dvw;
  max-width: 653px;
  height: 3rem;
  margin: 0.3rem 0;
  border: none;
  border-radius: 10rem;
  font-size: 1rem;
  color: var(--black);
}

.lower-buttons button:hover {
  cursor: pointer;
  border: solid var(--main-green) 0.1rem;
	box-shadow: 0.25em 0.25em .75em -.5em var(--shadow-color);
}

.lower-buttons
.button-home {
  background-color: var(--white);
}

.back-button {
  width: 3rem;
  height: 3rem;
  margin: 0.3rem 0;
  padding-top: 0.2rem;
  border: none;
  border-radius: 10rem;
  font-size: 1.2rem;
}

.back-button:hover {
  cursor: pointer;
  background-color: var(--main-green);
  color: var(--white);
}

/* Phones */
@media only screen and (max-width: 600px) {
  .main-container {
    grid-column-start: 1;
  }
}

/* Portrait tablets and large phones */
@media only screen and (min-width: 601px) and (max-width: 1024px) {
  .main-container {
    grid-column-start: 2;
  }
}

/* Small laptops and desktops */
@media only screen and (min-width: 1025px) {
  .main-container {
    grid-area: 1 / 2 / span 10 / span 1;
  }

  .lower-buttons button {
    width: 30dvw;
  }
}
