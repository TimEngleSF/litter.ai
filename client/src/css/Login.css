.login-container {
  width: 100%;
}

.login-logo {
  display: flex;
  width: 100%;
  height: 30dvh;
  background-color: var(--main-green);
  margin-bottom: 1.5rem;
  align-items: center;
  justify-content: center;
}

.login-logo-image{
  max-height: 10rem;
  grid-column-start: 2;
  grid-row-start: 2;
  object-fit: contain;
}

.login-form {

}

.invalid-login {
  color: rgb(255, 0, 0);
  text-align: center;
}

.login-email, .login-password {
  width: 50%;;
  margin: 10px auto;
}

.login-email label, .login-password label {
  color: gray;
  font-size: small;
}

.login-email input, .login-password input {
  display: block;
  width: 100%;
}