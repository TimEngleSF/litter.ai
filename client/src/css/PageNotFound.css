.PNF-container {
  height: 100dvh;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.PNF-container
.PNF-404 {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
}

.PNF-container
.PNF-404
.PNF-icon {
  font-size: 10rem;
  position: absolute;
  color: var(--main-green)
}

.PNF-container
.PNF-404 h1 {
  font-size: 11rem;
  margin: 0;
}

.PNF-container h3 {
  margin: -1rem 0 5rem 0;
}

.PNF-container
.PNF-buttons button {
  width: 15dvw;
}

/* Phones */
@media only screen and (max-width: 600px) {

}

/* Portrait tablets and large phones */
@media only screen and (min-width: 601px) and (max-width: 1024px) {

}

/* Small laptops and desktops */
@media only screen and (min-width: 1025px) {

}