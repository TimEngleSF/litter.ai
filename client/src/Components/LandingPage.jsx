import React from 'react';
import { PiFlowerTulipThin } from 'react-icons/pi';
import { Link } from 'react-router-dom';
import '../css/Landing.css';

export default function LandingPage({ user }) {
  return (
    <>
      <div className="landing-container main-container">
        <div className="landing-top-section">
          <PiFlowerTulipThin className="landing-logo" />
          <div className="landing-text">
            <h1>Welcome to LitterSort</h1>
            <p>With just a photo, sort your garbage and save the world.</p>
          </div>
          <div className="landing-top-buttons lower-buttons">
            <Link to="/capture">
              <button type="button" className="landing-capture-button">
                Capture Picture
              </button>
            </Link>
          </div>
        </div>
        <div className="landing-bottom-section">
          <div className="landing-bottom-buttons lower-buttons">
            <Link to="/leaderboard">
              <button type="button" className="landing-secondary-button">
                Leaderboard
              </button>
            </Link>
            { user
              ? (
                <Link to="/profile">
                  <button type="button" className="landing-secondary-button">
                    Profile
                  </button>
                </Link>
              )
              : (
                <>
                  <Link to="/register">
                    <button type="button" className="landing-secondary-button">
                      Sign Up
                    </button>
                  </Link>
                  <div className="landing-login-text">
                    <p>Already have an account?</p>
                    <Link to="/login">
                      Log in
                    </Link>
                  </div>
                </>
              )}
          </div>
        </div>
      </div>
      <div className="landing-full-bg" />
    </>
  );
}
