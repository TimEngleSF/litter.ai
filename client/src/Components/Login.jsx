import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { FaAngleLeft, FaRegEnvelope } from 'react-icons/fa6';
import { FiLock } from 'react-icons/fi';
import { fetchLogin } from '../utils/fetchUserData';
import '../css/LoginSignUp.css';

export default function Login({ setUser }) {
  const [loginData, setLoginData] = useState({
    email: '',
    password: '',
  });

  const [invalidLogin, setInvalidLogin] = useState(false);
  const navigate = useNavigate();

  const handleLogin = async () => {
    try {
      const response = await fetchLogin(loginData);
      setUser(response.user.displayUsername);
      setInvalidLogin(false);
      navigate('/');
    } catch (err) {
      setInvalidLogin(true);
      console.error(err);
    }
  };

  const handleChange = (e) => {
    e.preventDefault();
    setLoginData({ ...loginData, [e.target.id]: e.target.value });
  };

  const goBack = () => {
    navigate(-1);
  };

  return (
    <div className="login-signup-container main-container">
      <div className="login-signup-header">
        <button className="back-button" type="button" aria-label="Back" onClick={goBack}>
          <FaAngleLeft />
        </button>
        <div className="login-signup-header-text">
          <h3>Sign In</h3>
          <p>Enter your credentials to continue</p>
        </div>
      </div>

      <form className="login-signup-form">
        {invalidLogin
          ? (
            <div className="invalid-login-signup">
              <small>Incorrect username and/or password</small>
            </div>
          )
          : <br className="invalid-login-signup" />}
        <div className="login-signup-email">
          <FaRegEnvelope className="login-signup-icon" />
          <input
            type="email"
            id="email"
            aria-label="Email"
            placeholder="Email"
            value={loginData.email}
            onChange={(e) => handleChange(e)}
          />
        </div>
        <br className="login-signup-form-break" />
        <div className="login-signup-password">
          <FiLock className="login-signup-icon" />
          <input
            type="password"
            id="password"
            aria-label="Password"
            placeholder="Password"
            minLength="4"
            value={loginData.password}
            onChange={(e) => handleChange(e)}
          />
        </div>
      </form>

      <div className="lower-buttons login-signup-buttons">
        <button
          type="button"
          onClick={handleLogin}
          id="login-signup-button"
        >
          Login
        </button>
        <div className="sign-up">
          <small>Don&apos;t have an account?</small>
          <Link
            to="/register"
            className="register-button"
          >
            <small>Sign Up</small>
          </Link>
        </div>
      </div>
    </div>
  );
}
