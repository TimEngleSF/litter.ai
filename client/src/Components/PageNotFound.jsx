import React from 'react';
import { Link } from 'react-router-dom';
import { MdOutlineWrongLocation } from 'react-icons/md';
import '../css/PageNotFound.css';

export default function PageNotFound() {
  return (
    <div className="PNF-container main-container">
      <div className="PNF-404">
        <MdOutlineWrongLocation className="PNF-icon" />
        <h1>4&nbsp;&nbsp;&nbsp;&nbsp;4</h1>
      </div>
      <h3>Your litter will have to be recycled elsewhere.</h3>
      <div className="PNF-buttons lower-buttons">
        <Link to="/">
          <button type="button">
            Home
          </button>
        </Link>
      </div>
    </div>
  );
}
