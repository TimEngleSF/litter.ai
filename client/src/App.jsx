import React, { useState, useEffect } from 'react';
import './css/App.css';
import Cookies from 'js-cookie';
import {
  BrowserRouter as
  Router,
  Routes,
  Route,
} from 'react-router-dom';
import categoryData from './MockData/mockCategoryData';
import LandingPage from './components/LandingPage';
import CameraCapture from './components/CameraCapture';
import Leaderboard from './components/Leaderboard/Leaderboard';
import Profile from './components/Profile/Profile';
import SuccessfulSubmission from './components/SuccessfulSubmission/SuccessfulSubmission';
import Register from './components/Register/Register';
import Login from './components/Register/Login';
import UserContextProvider from './utils/UserContextProvider';

export default function App() {
  const [userData, setUserData] = useState({});

  useEffect(() => {
    const user = Cookies.get('userData');
    setUserData(JSON.parse(user));
  }, []);

  return (
    <div className="App">
      <UserContextProvider>
        <Router>
          <Routes>
            <Route path="/" element={<LandingPage />} />
            <Route path="/capture" element={<CameraCapture />} />
            <Route path="/leaderboard" element={<Leaderboard />} />
            <Route path="/profile" element={<Profile user={userData} />} />
            <Route path="/register" element={<Register />} />
            <Route path="/login" element={<Login user={userData} />} />
            <Route path="/success" element={<SuccessfulSubmission type={categoryData.plastic} />} />
          </Routes>
        </Router>
      </UserContextProvider>
    </div>
  );
}
