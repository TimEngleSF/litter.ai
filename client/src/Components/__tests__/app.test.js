/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  act,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { MemoryRouter as Router, Routes, Route } from 'react-router-dom';
import App from '../../App';
import PageNotFound from '../PageNotFound';
import Login from '../Login';
import LandingPage from '../LandingPage';

describe('Page Not Found component', () => {
  const PNFpage = <Router><PageNotFound /></Router>;
  // Creates a snapshot
  test('Leaderboard matches the current snapshot', () => {
    const tree = renderer.create(
      <Router>
        <PageNotFound />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('404 page exists', () => {
    render(PNFpage);
    expect(screen.getByTestId('PNF-icon')).toBeInTheDocument();
    expect(screen.getByText('Your litter will have to be recycled elsewhere.')).toBeInTheDocument();
  });

  test('Home button redirects to the home page', () => {
    /* The Routes component is erroring and only renders the LandingPage as
    a Route, putting PageNotFound outside of Routes is a temporary solution */
    render(
      <Router>
        <PageNotFound />
        <Routes>
          <Route path="/" element={<LandingPage />} />
        </Routes>
      </Router>,
    );
    expect(screen.getByTestId('PNF-icon')).toBeInTheDocument();
    expect(screen.getByText('Your litter will have to be recycled elsewhere.')).toBeInTheDocument();

    fireEvent.click(screen.getByRole('button', { name: 'Home' }));
    expect(screen.getByText('Welcome to LitterSort')).toBeInTheDocument();
  });
});
