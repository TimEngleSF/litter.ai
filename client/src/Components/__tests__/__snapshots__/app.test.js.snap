// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Not Found component Leaderboard matches the current snapshot 1`] = `
<div
  className="PNF-container main-container"
>
  <div
    className="PNF-404"
  >
    <svg
      className="PNF-icon"
      data-testid="PNF-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M18 11v.2c0 2.34-1.95 5.44-6 9.14-4.05-3.7-6-6.79-6-9.14C6 7.57 8.65 5 12 5c.34 0 .68.03 1 .08V3.06c-.33-.04-.66-.06-1-.06-4.2 0-8 3.22-8 8.2 0 3.32 2.67 7.25 8 11.8 5.33-4.55 8-8.48 8-11.8V11h-2z"
      />
      <circle
        cx="12"
        cy="11"
        r="2"
      />
      <path
        d="M22.54 2.88l-1.42-1.42L19 3.59l-2.12-2.13-1.42 1.42L17.59 5l-2.13 2.12 1.42 1.42L19 6.41l2.12 2.13 1.42-1.42L20.41 5z"
      />
    </svg>
    <h1>
      4    4
    </h1>
  </div>
  <h3>
    Your litter will have to be recycled elsewhere.
  </h3>
  <div
    className="PNF-buttons lower-buttons"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        type="button"
      >
        Home
      </button>
    </a>
  </div>
</div>
`;
